Wikipedia/Code/WMFContinueReadingContentSource.m

```diff
@@ -35,11 +35,13 @@ - (void)stopUpdating {
 
 - (void)loadNewContentInManagedObjectContext:(NSManagedObjectContext *)moc force:(BOOL)force completion:(nullable dispatch_block_t)completion {
     NSURL *lastRead = [[NSUserDefaults wmf] wmf_openArticleURL] ?: self.userDataStore.historyList.mostRecentEntry.URL;
+    if (!lastRead) {
+        return;
```

Comment:
Missing a call to the completion here - causing the hang on launch @tonisevener described